engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/IOMappingHelper.java

```diff
@@ -45,34 +48,56 @@
       variablesState.removeTemporaryVariables(elementInstanceKey);
     }
 
-    if (hasOutputMappings) {
-      mergeTool.reset();
-
-      final DirectBuffer variables =
-          determineVariables(variablesState, elementInstanceKey, outputMappings);
-
-      mergeTool.mergeDocumentStrictly(variables, outputMappings);
-      final DirectBuffer mergedVariables = mergeTool.writeResultToBuffer();
+    if (!outputMappings.isEmpty()) {
+      final DirectBuffer mergedVariables =
+          mergeOutputVariables(variablesState, elementInstanceKey, outputMappings);
 
       variablesState.setVariablesFromDocument(
           getVariableScopeKey(context), workflowKey, mergedVariables);
     }
   }
 
+  private DirectBuffer mergeOutputVariables(
```

Comment:
I'm wondering why we haven't found this issue before :sweat_smile:

Reply:
Yeah :see_no_evil: